internal/constraints/cache.go

```diff
@@ -170,3 +199,20 @@ func (c *Cache) getExpectedConstraintDescriptor(
 		return expected, ok
 	}
 }
+
+func reparseUnrecognized(
```

Comment:
We [added a resolver to this function](https://github.com/bufbuild/buf/pull/3317/files#diff-85d4d81d3a6dfea74ad1d6a24a3bbc1654253cd5632e8e0fe3ae7618433bf2d0R78-R82) with synchronicity and same parameter name as well 😄

Reply:
but yea i'm guessing I don't need the copy of the function any more since I should be to a resolver to create the validator. Either way, it's not a concern of this PR.